ddl/table.go

```diff
@@ -336,6 +353,30 @@ func (w *worker) onRecoverTable(d *ddlCtx, t *meta.Meta, job *model.Job) (ver in
 			}
 		})
 
+		var newRules []*label.Rule
+		for _, r := range oldRules {
+			if r.ID == tableRuleID {
+				newRules = append(newRules, r.Clone().Reset(tblInfo.ID, job.SchemaName, tblInfo.Name.L))
+			}
+		}
+
+		if tblInfo.GetPartitionInfo() != nil {
+			for _, r := range oldRules {
+				for _, def := range tblInfo.GetPartitionInfo().Definitions {
```

Comment:
It does not conflict. You are constructing strings from scractch for every partition. It is reasonable to avoid such behavior since there may be tons of partitions. Moreover, the construction is in another loop, thus will execute `n^2` times or so.

It works, but not in the most efficient way.